src/common/download.h

```diff
@@ -32,5 +32,11 @@
 
 namespace tools
 {
-  bool download(const std::string &path, const std::string &url);
+  typedef void *download_async_handle;
```

Comment:
What is the difference between
using async_handle_handle =
  std::unique_ptr<download_async_handle_impl, download_async_cleanup>;
and a typedef ?